cordova-lib/src/cordova/platform.js

```diff
@@ -174,6 +181,24 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
     });
 }
 
+function save(hooksRunner, projectRoot, opts) {
+    var xml = cordova_util.projectConfig(projectRoot);
+    var cfg = new ConfigParser(xml);
+
+    // First, remove all platforms that are already in config.xml
+    cfg.getEngines().map(function(engine){
+        cfg.removeEngine();
```

Comment:
Something strange here. `removeEngine` requires an engine Id param (https://github.com/apache/cordova-lib/blob/master/cordova-lib/src/configparser/ConfigParser.js#L404). Probably should be `cfg.removeEngine(engine.id);` instead.
Also `forEach` IMHO is more appropriate here than `map` here